app/Http/Controllers/CampaignsController.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace Rogue\Http\Controllers;
+
+use Rogue\Models\Campaign;
+use Illuminate\Http\Request;
+use Rogue\Http\Transformers\CampaignTransformer;
+
+class CampaignsController extends ApiController
+{
+    /**
+     * @var Rogue\Http\Transformers\SignupTransformer;
+     */
+    protected $transformer;
+
+    /**
+     * Create a controller instance.
+     *
+     */
+    public function __construct()
+    {
+        $this->transformer = new CampaignTransformer;
+
+        // $this->middleware('auth');
+        // $this->middleware('role:admin,staff', ['only' => ['store', 'update', 'destroy']]);
+        // $this->middleware('scopes:write', ['only' => ['store', 'update', 'destroy']]);
+    }
```

Comment:
I don't think we need the `auth` middleware but we should require that someone is an `admin`/`staff` and has the `write` scope to be able to create a campaign, update a campaign, or delete a campaign. However, I keep getting `403` status code and the `You don't have the correct role to do that!` message when I comment back in lines 25 and 26 but I checked my token to confirm the role is `admin` and has the `write` scope... is there something I'm missing?

Reply:
@chloealee what does `auth` do? Makes sure someone is logged in?

Reply:
oh yes that is what it is @katiecrane ! Ok so I think we do need `auth` then but only for `store`, `update`, and `destroy`, same with the other middleware but still can't get this to work :-/ Now I'm getting `"message": "Unauthenticated."`

Reply:
@chloealee were you able to actually hit the endpoint though?

Reply:
@katiecrane if I comment out the middleware, the logic works in Postman! I'm not sure if this is sufficient but I was planning on fully testing this when I build out the front end and via a "Create Campaign" button.

Reply:
@chloealee hmm in the meantime would that block a prod deploy?

Reply:
@katiecrane I'm don't think so because nothing is really using it yet? But this can totally also wait to be deployed with the front end when that is done too if that makes more sense and so we can totally test this works with 💯 percent confidence!

Reply:
@chloealee you're gonna need it to do the front end part though! I think it's okay to leave a note in here about what might not be working/needs to be tested since this whole campaigns thing is a WIP, as long as we can still deploy to prod